### DISCUSSION TITLE
ValueError: Unable to configure handler 'netbox_auth_log'

### DISCUSSION THREAD

**Discussion by simo735** [netbox-community/netbox#8965] [Q&A] [answered]
Hello, 

I am trying to add logging to Netbox but i got those error:

> return _bootstrap._gcd_import(name[level:], package, level)
> File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
> File "<frozen importlib._bootstrap>", line 983, in _find_and_load
> File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
> File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
> File "<frozen importlib._bootstrap_external>", line 728, in exec_module
> File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
> File "/opt/netbox/netbox/netbox/wsgi.py", line 7, in <module>
> application = get_wsgi_application()
> File "/opt/netbox-3.1.5/venv/lib/python3.7/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
> django.setup(set_prefix=False)
> File "/opt/netbox-3.1.5/venv/lib/python3.7/site-packages/django/__init__.py", line 19, in setup
> configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
> File "/opt/netbox-3.1.5/venv/lib/python3.7/site-packages/django/utils/log.py", line 75, in configure_logging
> logging_config_func(logging_settings)
> File "/usr/local/lib/python3.7/logging/config.py", line 800, in dictConfig
> dictConfigClass(config).configure()
> File "/usr/local/lib/python3.7/logging/config.py", line 571, in configure
> '%r' % name) from e
> ValueError: Unable to configure handler 'netbox_auth_log'
> [INFO] Worker exiting (pid: 31660)
> [WARNING] Worker with pid 31656 was terminated due to signal 15
> [WARNING] Worker with pid 31657 was terminated due to signal 15
> [WARNING] Worker with pid 31658 was terminated due to signal 15
> [WARNING] Worker with pid 31659 was terminated due to signal 15
> [INFO] Shutting down: Master
> [INFO] Reason: Worker failed to boot.
> netbox.service: main process exited, code=exited, status=3/NOTIMPLEMENTED
> Unit netbox.service entered failed state.
> netbox.service failed.

The configuration that i use is this:

```
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'netbox_auth_log': {
            'level': 'DEBUG',
            'class': 'logging.handlers.RotatingFileHandler',
            'filename': '/opt/netbox/logs/django-ldap-debug.log',
            'maxBytes': 1024 * 500,
            'backupCount': 5,
        },
    },
    'loggers': {
        'django_auth_ldap': {
            'handlers': ['netbox_auth_log'],
            'level': 'DEBUG',
        },
    },
}
```

I tried also the default one in the documentation but i got the some errors.
If i remove the logging configuration Netbox start without issue again:

```
LOGGING = {}
```

**Comment by jeremystretch** [comment, marked as answer]
Does the file `/opt/netbox/logs/django-ldap-debug.log` exist, and does the NetBxox system user have write access to it? With the file in place, your logging config seems to work fine for me.

**Reply by simo735** [reply]
Thanks Jeremy. the NetBox system user write access was missing. 
I add and the application start well.

**Reply by jeremystretch** [reply]
It's not a very helpful error, is it? :thinking: Glad you got it sorted!
